backend/rules/types/types.go

```diff
@@ -21,6 +22,10 @@ type Result struct {
 	Context      string
 }
 
+type GraphPathResult struct {
```

Comment:
unneeded layer of indirection

Reply:
Passing in lists of paths now. Since we can have different sets of paths. Easier to process.